### PR TITLE
Use sdk v2 in hybrid apps

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

Migrate hybrid apps examples to v2.
For now we still use the same core as using 2 cores would lead to separate sessions for native and RN.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [ ] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

File: android-app-with-rn/android/app/build.gradle [hunk 1]
```diff
@@ -11,7 +11,7 @@ datadogProperties.load(new FileInputStream(datadogPropertiesFile))
 
 android {
     namespace 'com.example.hybridapp'
-    compileSdk 32
+    compileSdk 33
```
Comment (0xnm) [inline android-app-with-rn/android/app/build.gradle:14]: The latest one is 34. It is worth to update `targetSdk` below as well.

Also it makes sense to bump JVM target from 1.8 to 11 in the configuration below (SDK v2 anyway is shipped with JVM 11 bytecode).

File: android-app-with-rn/android/app/build.gradle [hunk 2]
```diff
@@ -63,6 +66,15 @@ dependencies {
     testImplementation 'junit:junit:4.13.2'
     androidTestImplementation 'androidx.test.ext:junit:1.1.3'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+
+    constraints {
```
Comment (0xnm) [inline android-app-with-rn/android/app/build.gradle:70]: what causes the need to add this? normally if all libs are resolved to `1.8`, there shouldn't be a need for that.

Also it is maybe worth to use the latest patch release for `1.8` (`1.8.22`, I guess).

File: android-app-with-rn/android/app/src/main/java/com/example/hybridapp/MyReactApplication.kt [hunk 3]
```diff
@@ -25,22 +23,21 @@ class MyReactApplication : Application(), ReactApplication {
         super.onCreate()
 
         val configuration = Configuration.Builder(
-            logsEnabled = true,
-            tracesEnabled = true,
-            crashReportsEnabled = true,
-            rumEnabled = true
+            clientToken = BuildConfig.CLIENT_TOKEN,
+            env = BuildConfig.ENVIRONMENT,
+            variant = "prod"
```
Comment (0xnm) [inline android-app-with-rn/android/app/src/main/java/com/example/hybridapp/MyReactApplication.kt:28]: Is it worth to set it, given there are no actually variants?

File: android-app-with-rn/android/app/src/main/java/com/example/hybridapp/MyReactApplication.kt [hunk 4]
```diff
@@ -25,22 +23,21 @@ class MyReactApplication : Application(), ReactApplication {
...
+        ).build()
+        Datadog.initialize(this, configuration, TrackingConsent.GRANTED)
+        Datadog.setVerbosity(Log.VERBOSE)
```
Comment (0xnm) [inline android-app-with-rn/android/app/src/main/java/com/example/hybridapp/MyReactApplication.kt:31]: by changing the order of lines we can get logs during initialization

```suggestion
        Datadog.setVerbosity(Log.VERBOSE)
        Datadog.initialize(this, configuration, TrackingConsent.GRANTED)
```

File: android-app-with-rn/android/app/build.gradle (hunk 2 above)
Comment (louiszawadzki) [reply android-app-with-rn/android/app/build.gradle:70]: I had the usual Kotlin compiling issue when not adding this bit.
Can we solve this in a different way?

**Review by 0xnm — APPROVED** [review]

**Pull request merged**